### PR TITLE
Add environment variables to the Dart script.

### PR DESCRIPTION
**Pull request by nex3**
This makes it possible for custom scripts to detect which Dart tasks are
running.

### CODE REVIEW

File: lib/travis/build/script/dart.rb [hunk 1]
```diff
@@ -71,6 +71,9 @@ def configure
         def export
           super
 
+          sh.export 'TRAVIS_DART_TEST', (!!task[:test]).to_s, echo: false
```
Comment (BanzaiMan) [inline lib/travis/build/script/dart.rb]: This would result in `export TRAVIS_DART_TEST=true`. Is that intended?

Comment (nex3) [inline lib/travis/build/script/dart.rb]: Yes (as long as the current task has a `test` field).

**Comment by BanzaiMan** [comment]
Please open a documentation PR as well. Thank you!

**Comment by nex3** [comment]
Can you remind me where documentation goes?

**Comment by BanzaiMan** [comment]
Sure. It goes to https://github.com/travis-ci/docs-travis-ci-com.

**Comment by nex3** [comment]
Done in https://github.com/travis-ci/docs-travis-ci-com/pull/1223

**Pull request merged**